pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -66,6 +66,22 @@ var appPersistPaths = []string{
 	types.AppImgDirname,
 }
 
+func encodeErrorInfo(errStr string, errTime time.Time) *info.ErrorInfo {
+	if errTime.IsZero() {
+		// No Success / Error to report
+		return nil
+	}
+	// Timestamp set. Even if the error is "" - the timestamp indicates the last
+	//  testing time. Worth sending it.
+	errInfo := new(info.ErrorInfo)
+	errInfo.Description = errStr
+	protoTime, err := ptypes.TimestampProto(errTime)
+	if err != nil {
```

Comment:
This begs the question (avoided in the original code) whether we should log.Error if we get an error back.

Reply:
Good point.. Will add it.

Reply:
Done.